### DISCUSSION TITLE
How much data is loaded?

### DISCUSSION THREAD

**Discussion by nbrown02** [nbrown02/FlowViz#62] [Q&A] [answered]
If you have a project/organisation that has been in use for many years (or migrated from TFS) this might be a question you have...

**Comment by nbrown02** [comment, marked as answer]
For each Work Item table in the dataset, the amount of data is as follows:

Last 6 months of data:
- WorkItems Bl0cked
- WorkItems Blocked
- WorkItems FlowEfficiency
- WorkItems Blocked2
- WorkItems Daily Blocked Count
- WorkItems WIP Age
(Note: this uses 6 months as they query any changes to any fields on work items, thus can be quite large)

Last 12 months of data (from the date of refresh)
- WorkItems Completed
- WorkItems In Progress
- WorkItems In Progress2

If you'd like to change this to use more/less data, you need to go into Power Query Editor, go to the Source and change the highlighted number to be the number of months worth of data you'd like to query

![](https://raw.githubusercontent.com/nbrown02/FlowViz/main/Screenshots/Query%20Range.png)

**Reply by angelo-i** [reply]
Might be interesting to parametrize this... or ask for new values when a user opens the template for 1st time.

**Reply by nbrown02** [reply]
Yeah it's something I've considered in the past, just not had enough users request it :)

**Reply by ghost** [reply]
Has the way dates are determined changed since this discussion? All I see are hard-coded dates in the sources. For example:
<img width="1175" alt="image" src="https://github.com/user-attachments/assets/0ba43c2b-d133-48a5-92f4-f7f0842bd14a" />

**Reply by nbrown02** [reply]
Hey @KodiakScott - unfortunately parameterizing it means when it's published to the web service, users can't schedule refreshes (for some reason Power BI doesn't allow it), so I reverted the implementation of it some years back.

**Reply by ghost** [reply]
Oh, interesting. Thank you for the quick response. I did see the WIki item about how much data is brought in after I asked this question.
